### PR TITLE
Update release docs

### PR DESCRIPTION
**Pull request by palango**
**Thank you for submitting this pull request :)**

**Short description**

Improve release docs.

**Definition of Done**

- [ ] Steps to manually test the change have been documented
- [ ] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1.
2.

### CODE REVIEW

**Review by weilbith — APPROVED** [review]
Cool thanks that you did that. One of the reasons I wanted you to do the release this time. :+1:

File: RELEASE.md [hunk 1]
```diff
@@ -54,20 +56,20 @@ Go to the `raiden-ts` directory to start bump the package version.
...
 
-Then run `pnpm version` to bump the version for`package.json`. Let's assume that we prepare for a new release containing a couple of minor bug fixes. You have to run the following command: 
+Then run `pnpm version` to bump the version for`package.json`. Let's assume that we prepare for a new release containing a couple of minor bug fixes. You have to run the following command:
 
 ```bash
 pnpm version patch
 ```
 
-Assuming `raiden-ts` was at version `0.22.0`, the command will update the package files' version tag to `0.22.1`. 
+Assuming `raiden-ts` was at version `0.22.0`, the command will update the package files' version tag to `0.22.1`.
```
[the remark this answers is not in the document]
Comment (palango) [reply RELEASE.md:65]: I was surprised that GitHub does display trailing whitespace by default. 

I'm not sure if this is worth adding linters and stuff. From time to time someone will edit this with an editor that removes it.

[the remark this answers is not in the document]
Comment (weilbith) [reply RELEASE.md:65]: Hmm. I disagree but that is fine for this PR. :grimacing:

File: RELEASE.md [hunk 2]
```diff
@@ -30,20 +28,24 @@ In preparation for the release changes create a new branch:
 git checkout -b prepare_release
 ```
 
-Please ensure that you also update the [CHANGELOG.md](./raiden-ts/CHANGELOG.md) accordingly. You should have the changes for the new version tracked as `[Unreleased]`. You only need to replace `[Unreleased]` with the new version. 
+Please ensure that you also update the [CHANGELOG.md](./raiden-ts/CHANGELOG.md) accordingly. You should have the changes for the new version tracked as `[Unreleased]`. You only need to replace `[Unreleased]` with the new version. Also make sure to add the link to the diff between the current and the new version.
```
[the remark this answers is not in the document]
Comment (palango) [reply RELEASE.md:31]: Good point.

File: RELEASE.md [hunk 3]
```diff
@@ -30,20 +28,24 @@ In preparation for the release changes create a new branch:
...
+
+```md
+[0.12.0]: https://github.com/raiden-network/light-client/compare/v0.11.1...v0.12.0
```
[the remark this answers is not in the document]
Comment (palango) [reply RELEASE.md:34]: I added a diff instead. Should make things clearer.

File: RELEASE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (palango) [reply RELEASE.md:69]: DO you know a good markdown linter?

**Pull request merged**